src/legacy/status_im/chat/models/mentions.cljs

```diff
@@ -100,29 +103,53 @@
              (assoc-in [:chats/mentions chat-id :mentions] state)
              (assoc-in [:chat/inputs-with-mentions chat-id] input-segments))}))
 
+(defonce current-call-id (atom 0))
+
 (rf/defn on-change-text
   {:events [:mention/on-change-text]}
-  [{:keys [db]} text]
+  [{:keys [db now]} text]
   (let [chat-id (:current-chat-id db)
-        params  [chat-id text]
+        call-id (inc (get-in db [:chat/inputs chat-id :current-call-id] 0))
+        params  [chat-id text call-id now]
         method  "wakuext_chatMentionOnChangeText"]
     (log/debug "[mentions] on-change-text" {:params params})
-    {:json-rpc/call [{:method     method
+    {:db            (assoc-in db [:chat/inputs chat-id :current-call-id] call-id)
+     :json-rpc/call [{:method     method
                       :params     params
                       :on-success #(rf/dispatch [:mention/on-change-text-success %])
                       :on-error   #(rf/dispatch [:mention/on-error
                                                  {:method method
                                                   :params params} %])}]}))
-
 (rf/defn on-change-text-success
   {:events [:mention/on-change-text-success]}
-  [{:keys [db]} result]
+  [{:keys [db now]} result]
   (log/debug "[mentions] on-change-text-success" {:result result})
-  (let [{:keys [state chat-id mentionable-users input-segments]} (transfer-mention-result result)]
-    {:db (-> db
-             (assoc-in [:chats/mention-suggestions chat-id] mentionable-users)
-             (assoc-in [:chats/mentions chat-id :mentions] state)
-             (assoc-in [:chat/inputs-with-mentions chat-id] input-segments))}))
+  (let [{:keys [state chat-id mentionable-users
+                input-segments call-id call-time]} (transfer-mention-result result)
+        current-call-id                            (get-in db [:chat/inputs chat-id :current-call-id])]
+    (when (= call-id current-call-id)
+      {:db (-> db
+               (assoc-in [:chat/inputs-with-mentions chat-id] input-segments)
+               (assoc-in [:chats/mention-suggestions chat-id] mentionable-users)
+               (assoc-in [:chats/mentions chat-id :mentions] state))
+       :fx [[:dispatch
+             [:mention/re-render-text-input
+              {:chat-id           chat-id
+               :mentionable-users mentionable-users
+               :state             state
+               :input-segments    input-segments
+               :call-id           call-id
+               :response-time     (- now call-time)}]]]})))
+
+(rf/defn re-render-text-input
+  {:events [:mention/re-render-text-input]}
```

Comment:
It's better to use names of events that are not tied to component lifecycles. Re-rendering is a React concept.